### PR TITLE
PA-2170 Add ApprovedForSplOn to proceed to SPL

### PR DESCRIPTION
**Pull request by Fosol**
Require the user to select when the request to proceed to SPL was approved on.

image

### CODE REVIEW

**Review by agalbrai — APPROVED** [review]
Looks good!

File: frontend/src/features/projects/common/strings.ts [hunk 1]
```diff
@@ -33,7 +33,8 @@ export const onTransferredWithinTheGreTooltip =
 export const clearanceNotifictionSent =
   'Enter Clearance Notification Sent date to be able to Proceed to SPL or Not Include in SPL.';
 export const requestForSplReceivedOn =
-  'Enter the data the request to be added to SPL was received on.';
+  'Enter the date the request to be added to SPL was received on.';
```
Comment (jdtoombs) [inline frontend/src/features/projects/common/strings.ts:36]: `the request -> of the request` maybe?

**Review by jdtoombs — APPROVED** [review]

File: frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx [hunk 2]
```diff
@@ -108,79 +109,100 @@ const EnhancedReferralCompleteForm = ({
...
-                !formikProps.values.requestForSplReceivedOn
-              }
-              onClick={() => validateFormikWithCallback(formikProps, () => setProceedToSpl(true))}
-            >
-              Proceed to SPL
-            </Button>
-            {formikProps.values.statusCode !== ReviewWorkflowStatus.NotInSpl && (
-              <>
-                <OrText>OR</OrText>
+      {formikProps.values.workflowCode === 'ERP' && (
```
Comment (Fosol) [inline frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx:112]: Only show these dates and buttons if the current workflow is ERP.

File: frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx [hunk 3]
```diff
@@ -108,79 +109,100 @@ const EnhancedReferralCompleteForm = ({
...
+              outerClassName="col-md-2"
+              formikProps={formikProps}
+              disabled={isReadOnly}
+              field="requestForSplReceivedOn"
+            />
+          </Form.Row>
+          <Form.Row>
+            <Form.Label column md={4}>
+              SPL Addition Approved On
+              <TooltipIcon toolTipId="approvedForSplOn" toolTip={approvedForSplOn} />
```
Comment (Fosol) [inline frontend/src/features/projects/erp/forms/EnhancedReferralCompleteForm.tsx:129]: This is now required before proceeding to SPL

File: frontend/src/features/projects/common/strings.ts [hunk 4]
```diff
@@ -33,7 +33,8 @@ export const onTransferredWithinTheGreTooltip =
 export const clearanceNotifictionSent =
   'Enter Clearance Notification Sent date to be able to Proceed to SPL or Not Include in SPL.';
 export const requestForSplReceivedOn =
-  'Enter the data the request to be added to SPL was received on.';
+  'Enter the date of the request to be added to SPL was received on.';
+export const approvedForSplOn = 'Enter the date the request to be added to SPL was approved on.';
```
Comment (jdtoombs) [inline frontend/src/features/projects/common/strings.ts:37]: Sorry should have added a second comment :( same thing here

**Pull request merged**